route_mappings.go

```diff
@@ -295,21 +296,35 @@ func (a *App) buildRouteName(p string) string {
 
 	for index, part := range parts {
 
-		if strings.Contains(part, "{") || part == "" {
+		originalPart := parts[index]
+
+		var previousPart string
+		if index > 0 {
+			previousPart = parts[index-1]
+		}
+
+		var nextPart string
+		if len(parts) > index+1 {
+			nextPart = parts[index+1]
+		}
+
+		isIdentifierPart := strings.Contains(part, "{") && (strings.Contains(part, flect.Singularize(previousPart)))
+		isSimplifiedID, _ := regexp.MatchString(`{[\s+]?id[\s+]?}`, part)
```

Comment:
Regexes are expensive, can you pre-compile this one outside this scope?

Reply:
I will, thanks @stanislas-m !

Reply:
Changed it to use `{id}` string comparison.